src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -360,20 +437,23 @@ end
             # orthogonalization, see Hetmaniuk & Lehoucq, and Duersch et. al.
             # cP = copy(cX)
             # cP[Xn_indices,:] .= 0
-            e = zeros(eltype(X), size(cX, 1), M - prev_nlocked)
-            for i in 1:length(Xn_indices)
-                e[Xn_indices[i], i] = 1
-            end
+
+            lenXn = length(Xn_indices)
+            e = zero(similar(X, size(cX, 1), M - prev_nlocked))
```

Comment:
`zero(similar(` allocates twice. Should we have a `zeros_like` function or something, that could fallback to `zero(similar(` but be better optimized in specific cases?

Reply:
ping @vchuravy who might have an opinion

Reply:
Hm. OK then let's just have our own `zeros_like` function that does this, and later as we figure out the patterns we use we can design our own mini-API (eg something to replace comprehensions). Sucks there's no standard way to do this.